Sources/Datadog/RUM/RUMV2Configuration.swift

```diff
@@ -30,6 +33,45 @@ internal func createRUMConfiguration(configuration: FeaturesConfiguration.RUM) -
     )
 }
 
+internal struct RUMViewEventsFilter: EventsFilter {
+    let decoder: JSONDecoder
+
+    init(decoder: JSONDecoder = JSONDecoder()) {
+        self.decoder = decoder
+    }
+
+    func filter(events: [Event]) -> [Event] {
+        var seen = Set<String>()
+        var filtered = [Event]()
+
+        // reversed is O(1) and no copy because it is view on the original array
+        for event in events.reversed() {
```

Comment:
I would slightly favour functional implementation for the whole thing (slightly more difficult to break in the future)

Reply:
Also, we found the imperative `for ... in` being much faster for long sequences in [SR's text obfuscation](https://github.com/DataDog/dd-sdk-ios/pull/1024#discussion_r999054231). It doesn't apply here, as the number of events is rather dozens, not thousands - I'm just sharing findings :).